crates/ethereum/payload/src/lib.rs

```diff
@@ -278,37 +247,33 @@ where
             }
         }
 
-        // Configure the environment for the tx.
-        let tx_env = evm_config.tx_env(&tx);
-
-        let ResultAndState { result, state } = match evm.transact(tx_env) {
-            Ok(res) => res,
-            Err(err) => {
-                if let Some(err) = err.as_invalid_tx_err() {
-                    if err.is_nonce_too_low() {
-                        // if the nonce is too low, we can skip this transaction
-                        trace!(target: "payload_builder", %err, ?tx, "skipping nonce too low transaction");
-                    } else {
-                        // if the transaction is invalid, we can skip it and all of its
-                        // descendants
-                        trace!(target: "payload_builder", %err, ?tx, "skipping invalid transaction and its descendants");
-                        best_txs.mark_invalid(
-                            &pool_tx,
-                            InvalidPoolTransactionError::Consensus(
-                                InvalidTransactionError::TxTypeNotSupported,
-                            ),
-                        );
-                    }
-                    continue
+        let gas_used = match strategy
+            .execute_transaction(Recovered::new_unchecked(tx.tx(), tx.signer()))
```

Comment:
I think we should add this as helper functions to both

trait SignedTx and TransactionSigned
could you please open a good first issue for this?